src/index/api.rs

```diff
@@ -168,22 +161,144 @@ impl Api {
         )?
         .json()?;
 
-        let result = response
+        Ok(response
             .users
             .into_iter()
-            .filter(|data| {
-                !data
-                    .login
-                    .as_ref()
-                    .map(|login| login.is_empty())
-                    .unwrap_or_default()
-            })
-            .map(|data| CrateOwner {
-                avatar: data.avatar.unwrap_or_default(),
-                login: data.login.unwrap_or_default(),
-            })
-            .collect();
-
-        Ok(result)
+            .filter(|data| !data.login.is_empty())
+            .collect())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use mockito::mock;
+    use serde_json::{self, json};
+
+    fn api() -> Api {
+        Api::new(Some(Url::parse(&mockito::server_url()).unwrap())).unwrap()
+    }
+
+    #[test]
+    fn get_owners() {
+        let _m = mock("GET", "/api/v1/crates/krate/owners")
+            .with_header("content-type", "application/json")
+            .with_body(
+                serde_json::to_string(&json!({
+                "users": [
+                    {
+                        "id": 1,
+                        "login": "the_first_owner",
+                        "name": "name",
+                        "avatar": "http://something"
+                    },
+                    {
+                        "id": 2,
+                        "login": "the_second_owner",
+                        "name": "another name",
+                        "avatar": "http://anotherthing"
+                    }
+                ]}))
+                .unwrap(),
+            )
+            .create();
+
+        assert_eq!(
+            api().get_owners("krate").unwrap(),
+            vec![
+                GithubUser {
+                    id: 1,
+                    avatar: "http://something".into(),
+                    login: "the_first_owner".into()
+                },
+                GithubUser {
+                    id: 2,
+                    avatar: "http://anotherthing".into(),
+                    login: "the_second_owner".into()
+                }
+            ]
+        );
+    }
+
+    #[test]
+    fn get_release_info() {
+        let created = Utc::now();
+        let _m = mock("GET", "/api/v1/crates/krate/versions")
+            .with_header("content-type", "application/json")
+            .with_body(
+                serde_json::to_string(&json!({
+                "versions": [
+                    {
+                        "num": "1.2.3",
+                        "created_at": created.to_rfc3339(),
+                        "yanked": true,
+                        "downloads": 223,
```

Comment:
so we double check we can deserialize unknown fields
```suggestion
                        "downloads": 223,
                        "license": "MIT",
```